### PR TITLE
add: v2ex-news

### PR DESCRIPTION
**Pull request by chyroc**
## Description

view https://www.v2ex.com/ 

## Screencast

setting

notification

del-notification

plane-topic

topic

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @chyroc 

Thanks for your extension contribution 🔥 

Before I can review it properly, I have a few things that we need to address before continuing:

* At the moment, Raycast doesn't support localization and only supports US English. Please remove Chinese strings and replace if necessary with English.

* Could you move all screenshots to a folder called `media` instead and update the README

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

File: extensions/v2ex-news/package.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  "name": "v2ex-news",
+  "title": "v2ex-news",
+  "description": "show v2ex notifications, planes, topics and replies.",
+  "icon": "v2ex.png",
+  "author": "chyroc",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "plane",
+      "title": "Plane | 位面列表",
```
Comment (pernielsentikaer) [inline extensions/v2ex-news/package.json:12]:
```suggestion
      "title": "Browse Planes",
      "subtitle": "V2EX",
```

File: extensions/v2ex-news/package.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      "description": "show plane data",
+      "mode": "view"
+    },
+    {
+      "name": "profile",
+      "title": "Profile | 个人主页",
```
Comment (pernielsentikaer) [inline extensions/v2ex-news/package.json:18]:
```suggestion
      "title": "Show Profile",
      "subtitle": "V2EX",
```

File: extensions/v2ex-news/package.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  "title": "v2ex-news",
+  "description": "show v2ex notifications, planes, topics and replies.",
+  "icon": "v2ex.png",
+  "author": "chyroc",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "plane",
+      "title": "Plane | 位面列表",
+      "description": "show plane data",
```
Comment (pernielsentikaer) [inline extensions/v2ex-news/package.json:13]:
```suggestion
      "description": "Show plane data",
```

File: extensions/v2ex-news/package.json [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      "mode": "view"
+    },
+    {
+      "name": "profile",
+      "title": "Profile | 个人主页",
+      "description": "show your profile",
```
Comment (pernielsentikaer) [inline extensions/v2ex-news/package.json:19]:
```suggestion
      "description": "Show your profile",
```

File: extensions/v2ex-news/package.json [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      "mode": "view"
+    },
+    {
+      "name": "notification",
+      "title": "Notification | 通知",
+      "description": "show your notification",
```
Comment (pernielsentikaer) [inline extensions/v2ex-news/package.json:25]:
```suggestion
      "title": "Show Notification",
      "subtitle": "V2EX",
      "description": "show your notification",
```

File: extensions/v2ex-news/package.json [hunk 6]
```diff
@@ -0,0 +1,62 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "v2ex-news",
+  "title": "v2ex-news",
```
Comment (pernielsentikaer) [inline extensions/v2ex-news/package.json:4]:
```suggestion
  "title": "V2EX",
```
I think you should move the -news part (and rename extension folder)

File: extensions/v2ex-news/package.json [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+      "description": "show your notification",
+      "mode": "view"
+    }
+  ],
+  "preferences": [
+    {
+      "name": "token",
+      "required": true,
+      "title": "V2EX Token",
+      "placeholder": "input your v2ex token",
```
Comment (pernielsentikaer) [inline extensions/v2ex-news/package.json:34]:
```suggestion
      "placeholder": "input your V2EX token",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @chyroc 👋 

Thanks for the updates, much appreciated 🔥 

I have some feedback ready for you:

*  All three commands suffers of [Empty state flicker](https://developers.raycast.com/basics/prepare-an-extension-for-store#empty-states) - could you look into this?

* If wrong credentials, it keeps showing “Loading” and the spinner newer stops.

cJtxTHSnmL2jqfQl

What you could do is something like this:

PEOSBzQxILe0oUFz

or return an empty list (so it shows “no results”)

4dR6nV5SnlZrmVxu

Looking forward to testing it out again 😄 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

File: extensions/v2ex-news/package.json (hunk 6 above)
Comment (chyroc) [reply extensions/v2ex-news/package.json:4]: i changed the name , but folder name cannot change, because there has a folder named `v2ex` exist.

**Comment by chyroc** [comment]
@pernielsentikaer hi, i fix this pr, please review it again, thanks.

**Comment by pernielsentikaer** [comment]
Hi @chyroc 👋 

I'm not sure why I was not able to find the already existing extension for the same service when searching, that's my bad for not noticing right away.

We have a bit of a problem here, what we should do is that you should make a PR into the existing service, so we're extending that one. Does it make sense?

**Comment by chyroc** [comment]
@pernielsentikaer if you agree, i can change it , and , please let me also be owner of this v2ex extension

**Comment by pernielsentikaer** [comment]
Hi @chyroc 👋 

The author of the v2ex extension is @mopig – Let's see if we can catch him here 😄

I can't make you the owner, but we have support for “contributors” in package.json that's fully implemented

**Comment by mopig** [comment]
> 

What should I do for this PR?

**Comment by pernielsentikaer** [comment]
We're talking about extending your extension with some new features (as you can see in this thread) – would you mind that @mopig?

**Comment by mopig** [comment]
> 

I'm happy to have these new features, no problem at all.

**Comment by pernielsentikaer** [comment]
Awesome @mopig 💪

@chyroc that sounds like a plan to merge into the existing extension

**Comment by fe9lix** [comment]
Hey guys, I'm closing this because it sounds like this would need a new PR to integrate the changes to the other v2ex extension. Feel free to open a new PR with the contribution.

**Pull request closed without merging**